React/Base/RCTConvert.m

```diff
@@ -506,6 +506,207 @@ + (type)type:(id)json                                       \
+      @"systemBlueColor": @{},
+      @"systemBrownColor": @{ // iOS 13.0
+        RCTFallbackARGB: @(0xFFa2845e)
+      },
+      @"systemGreenColor": @{},
+      @"systemIndigoColor": @{ // iOS 13.0
+        RCTFallbackARGB: @(0xFF5856d6)
+      },
+      @"systemOrangeColor": @{},
+      @"systemPinkColor": @{},
+      @"systemPurpleColor": @{},
+      @"systemRedColor": @{},
+      @"systemTealColor": @{},
+      @"systemYellowColor": @{},
+      // Adaptable Gray Colors
+      @"systemGrayColor": @{},
+      @"systemGray2Color": @{ // iOS 13.0
+        RCTFallbackARGB: @(0xFFaeaeb2)
+      },
+      @"systemGray3Color": @{ // iOS 13.0
+        RCTFallbackARGB: @(0xFFc7c7cc)
+      },
+      @"systemGray4Color": @{ // iOS 13.0
+        RCTFallbackARGB: @(0xFFd1d1d6)
+      },
+      @"systemGray5Color": @{ // iOS 13.0
+        RCTFallbackARGB: @(0xFFe5e5ea)
+      },
+      @"systemGray6Color": @{ // iOS 13.0
+        RCTFallbackARGB: @(0xFFf2f2f7)
+      },
+#if DEBUG
+      // The follow exist for Unit Tests
+      @"unitTestFallbackColor": @{
+        RCTFallback: @"gridColor"
+      },
+      @"unitTestFallbackColorIOS": @{
+        RCTFallback: @"blueColor"
+      },
+      @"unitTestFallbackColorEven": @{
+        RCTSelector: @"unitTestFallbackColorEven",
+        RCTIndex: @0,
+        RCTFallback: @"controlAlternatingRowBackgroundColors"
+      },
+      @"unitTestFallbackColorOdd": @{
+        RCTSelector: @"unitTestFallbackColorOdd",
+        RCTIndex: @1,
+        RCTFallback: @"controlAlternatingRowBackgroundColors"
+      },
+#endif
+    };
+  }
+  return colorMap;
+}
+
+/** Returns a UIColor based on a semantic color name.
+ *  Returns nil if the semantic color name is invalid.
+ */
+static UIColor *RCTColorFromSemanticColorName(NSString *semanticColorName)
+{
+  NSDictionary<NSString *, NSDictionary *> *colorMap = RCTSemanticColorsMap();
+  UIColor *color = nil;
+  NSDictionary<NSString *, id> *colorInfo = colorMap[semanticColorName];
+  if (colorInfo) {
+    NSString *semanticColorSelector = colorInfo[RCTSelector];
+    if (semanticColorSelector == nil) {
+      semanticColorSelector = semanticColorName;
+    }
+    SEL selector = NSSelectorFromString(semanticColorSelector);
+    if (![UIColor respondsToSelector:selector]) {
+      NSNumber *fallbackRGB = colorInfo[RCTFallbackARGB];
+      if (fallbackRGB != nil) {
+        RCTAssert([fallbackRGB isKindOfClass:[NSNumber class]], @"fallback ARGB is not a number");
+        return [RCTConvert UIColor:fallbackRGB];
+      }
+      semanticColorSelector = colorInfo[RCTFallback];
+      selector = NSSelectorFromString(semanticColorSelector);
+    }
+    RCTAssert ([UIColor respondsToSelector:selector], @"RCTUIColor does not respond to a semantic color selector.");
+    Class klass = [UIColor class];
+    IMP imp = [klass methodForSelector:selector];
+    id (*getSemanticColorObject)(id, SEL) = (void *)imp;
+    id colorObject = getSemanticColorObject(klass, selector);
```

Comment:
hmm I am skeptical when it comes down to runtime invocation for these functions. My main concerns are:
1. `NSSelectorFromString`/`NSSelectorFromString` are slow methods
2. `respondsToSelector:`/`methodForSelector:selector` can really bite us in earlier OS versions that had these methods available but not publicly exposed. I definitely seen this happen in the past where a method was introduced on a later OS (from an API header perspective).

My general direction here would have been implementing the mapping function myself.
```
static UIColor colorForSemanticName(NSString *string)
{
  if (string isEqualToString:@"systemBlueColor"]){
    return [UIColor systemBlueColor];
  }
}
```
thoughts?

Reply:
We've been shipping this solution if Office for few months for iOS and even longer for macOS and haven't noticed perf problems.  For the matter of hitting undocumented selectors in earlier OS's: generally I agree with you, but worry less about semantic color accessors on UIColor/NSColor.   Maybe I'm wrong but it feels like an if/else chain of dozens of string compares would be slower than a NSSelectorFromString call.   I can Profile it.   Perhaps I could cache the selectors and/or IMPs.

Reply:
If that's a pattern we are confident with (using the runtime) I am fine with it. I generally tend to avoid this go down the if/else switch were I handle everything but I will leave this up to you.